src/service/private.overseas.entity.details.ts

```diff
@@ -18,15 +18,40 @@ export const getPrivateOeDetails = async (
     overseasEntityId,
   );
 
+  checkErrorResponse(req, response, overseasEntityId, transactionId, "overseas entity details");
+
+  return response.resource;
+};
+
+export const getBeneficialOwnerPrivateData = async (
+  req: Request,
+  transactionId: string,
+  overseasEntityId: string
+): Promise<BeneficialOwnersPrivateData | undefined> => {
+  const response = await makeApiCallWithRetry(
+    "overseasEntity",
+    "getBeneficialOwnerPrivateData",
+    req,
+    req.session as Session,
+    transactionId,
+    overseasEntityId
+  );
+
+  checkErrorResponse(req, response, overseasEntityId, transactionId, "beneficial owner details");
+
+  return response.resource;
+};
+
+const checkErrorResponse = (req: Request, response, overseasEntityId?: string, transactionId?: string, dataToRetrieve: string = "") => {
   if (response.httpStatusCode !== 200 && response.httpStatusCode !== 404) {
-    const errorMsg = `Something went wrong fetching overseas entity details = ${JSON.stringify(response)}`;
+    const errorMsg = `Something went wrong fetching private ${dataToRetrieve} details = ${JSON.stringify(response)}`;
     throw createAndLogErrorRequest(req, errorMsg);
   }
 
   if (response.httpStatusCode === 404) {
-    logger.debugRequest(req, `No overseas entity details found ${overseasEntityId} under ${transactionId}`);
+    logger.debugRequest(req, `No private ${dataToRetrieve} found for ${overseasEntityId} under ${transactionId}`);
     return undefined;
   }
 
-  return response.resource;
+  logger.debugRequest(req, `${response}`);
```

Comment:
revert this - after 1040 merges

Reply:
change it back to 'return response.resource'?